Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -0,0 +1,1155 @@
+                    currentGrabPoint = initialGrabPoint;
+                    initialScaleOnGrabStart = Target.transform.localScale;
+                    initialPositionOnGrabStart = Target.transform.position;
+                    grabPointInPointer = Quaternion.Inverse(eventData.Pointer.Rotation) * (initialGrabPoint - currentPointer.Position);
+
+                    // todo: move this out?
+                    SetHighlighted(grabbedHandleTransform);
+
+                    if (currentHandleType == HandleType.Scale)
+                    {
+                        // Will use this to scale the target relative to the opposite corner
+                        oppositeCorner = rigRoot.transform.TransformPoint(-grabbedHandle.transform.localPosition);
+                        diagonalDir = (grabbedHandle.transform.position - oppositeCorner).normalized;
+
+                        ScaleStarted?.Invoke();
+
+                        if (debugText != null)
+                        {
+                            debugText.text = "OnPointerDown:ScaleStarted";
+                        }
+                    }
+                    else if (currentHandleType == HandleType.Rotation)
+                    {
+                        currentRotationAxis = GetRotationAxis(grabbedHandleTransform);
+
+                        RotateStarted?.Invoke();
+
+                        if (debugText != null)
+                        {
+                            debugText.text = "OnPointerDown:RotateStarted";
+                        }
+                    }
+
+                    eventData.Use();
+                }
+            }
+
+            if (currentPointer != null)
+            {
+                // Always mark the pointer data as used to prevent any other behavior to handle pointer events
+                // as long as bounds control manipulation is active.
+                // This is due to us reacting to both "Select" and "Grip" events.
+                eventData.Use();
+            }
+        }
+
+        private void OnPointerDragged(MixedRealityPointerEventData eventData) { }
+
+        public void OnSourceDetected(SourceStateEventData eventData)
+        {
+            if (eventData.Controller != null)
+            {
+                if (sourcesDetected.Count == 0 || sourcesDetected.Contains(eventData.Controller) == false)
+                {
+                    sourcesDetected.Add(eventData.Controller);
+                }
+            }
+        }
+
+        public void OnSourceLost(SourceStateEventData eventData)
+        {
+            sourcesDetected.Remove(eventData.Controller);
+
+            if (currentPointer != null && currentPointer.InputSourceParent.SourceId == eventData.SourceId)
+            {
+                HandleType lastHandleType = currentHandleType;
+
+                currentPointer = null;
+                currentHandleType = HandleType.None;
+                // todo: move this out?
```

Comment:
Same here.

Reply:
i'll leave them for now and tackle them in my follow up PR - should be fine as we're still in a feature branch